components/raftstore/src/store/fsm/apply.rs

```diff
@@ -688,20 +689,21 @@ where
     }
 
     /// Flush all pending writes to engines.
-    /// If it returns true, all pending writes are persisted in engines.
-    pub fn flush(&mut self) -> bool {
-        // TODO: this check is too hacky, need to be more verbose and less buggy.
+    pub fn flush(&mut self) {
         let t = match self.timer.take() {
             Some(t) => t,
-            None => return false,
+            // This will lead to inaccurate delay calculation of write requests with uninitialized
```

Comment:
This is a bad idea. It will cause confusion for coprocessors if there is no writes at all.

Reply:
but timer.is_none() standing for no write is confusing as well.